### PR TITLE
Feature/124 error interceptor

### PR DESCRIPTION
**Pull request by Limbo2332**
https://github.com/BinaryStudioAcademy/bsa-2023-leetwars/assets/93576084/30c0be42-9128-4053-9ddc-81b9334084f5

### CODE REVIEW

**Comment by Limbo2332** [comment]
[Error interceptor](https://trello.com/c/J2B7tHe0/124-error-interceptor)

File: frontend/src/app/core/services/auth.service.ts [hunk 1]
```diff
@@ -51,16 +51,15 @@ export class AuthService {
             from(this.afAuth.createUserWithEmailAndPassword(user.email, user.password)).pipe(
                 first(),
                 tap(() => this.sendVerificationMail()),
-                catchError((error) => throwError(error.message)),
             ),
+            false,
             user.userName,
         );
```
Comment (SashaBondarchuk) [inline frontend/src/app/core/services/auth.service.ts]: false can be removed (see line 149)

File: backend/LeetWars.Core/LeetWars.Core.BLL/Services/UserService.cs [hunk 2]
```diff
@@ -49,16 +50,21 @@ public async Task<UserDto> CreateUserAsync(NewUserDto userDto)
...
+        if (isExistingUserName && userDto.IsWithProvider)
         {
             userDto.UserName = await GenerateUniqueUsername(userDto.Email);
         }
 
+        if(isExistingUserName && !userDto.IsWithProvider)
+        {
+            throw new InvalidUsernameOrPasswordException($"Error: This username is already registered in the system.");
+        }
+
```
Comment (SashaBondarchuk) [inline backend/LeetWars.Core/LeetWars.Core.BLL/Services/UserService.cs]:
```suggestion
if (isExistingUserName)
{
    userDto.UserName = userDto.IsWithProvider
        ? await GenerateUniqueUsername(userDto.Email)
        : throw new InvalidUsernameOrPasswordException($"Error: This username is already registered in the system.");
}

```

**Review by SashaBondarchuk — CHANGES_REQUESTED** [review]

File: frontend/src/app/core/services/auth.service.ts (hunk 1 above)
Comment (Limbo2332) [reply frontend/src/app/core/services/auth.service.ts]: Unreal to do, only changed to undefined

**Review by SashaBondarchuk — APPROVED** [review]

**Review by tatianahutii — APPROVED** [review]

**Review by Den313 — APPROVED** [review]

**Pull request merged**